Revit_Engine/Query/RevitParametersDifferences.cs

```diff
@@ -0,0 +1,141 @@
+using BH.oM.Adapters.Revit.Elements;
+using BH.oM.Adapters.Revit.Parameters;
+using BH.oM.Base;
+using BH.oM.Diffing;
+using BH.oM.Base.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace BH.Engine.Adapters.Revit
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Returns Property differences between RevitParameters (RevitPulledParameters/RevitParametersToPush) owned by the two input objects." +
+            "This method can be added as Func delegate to the DiffingConfig; if so, it is automatically triggered when Diffing the two objects.")]
+        [Input("obj1", "Past object being compared.")]
+        [Input("obj2", "Following object being compared.")]
+        [Input("comparisonConfig", "Comparison Config to be used during comparison.")]
+        [Output("parametersDifferences", "Differences in terms of RevitParametersToPush found on the two input objects.")]
+        public static List<PropertyDifference> RevitParametersDifferences<T>(this object obj1, object obj2, BaseComparisonConfig comparisonConfig) where T : class, IRevitParameterFragment
+        {
+            List<PropertyDifference> result = new List<PropertyDifference>();
+
+            Dictionary<string, RevitParameter> allParamsDict_obj1 = (obj1 as IBHoMObject)?.ParametersDictionary<T>();
+            Dictionary<string, RevitParameter> allParamsDict_obj2 = (obj2 as IBHoMObject)?.ParametersDictionary<T>();
+
+            if ((!allParamsDict_obj1?.Any() ?? true) || (!allParamsDict_obj2?.Any() ?? true))
+                return result; // no pulled parameters found.
+
+            result = ParameterDifferences(allParamsDict_obj1, allParamsDict_obj2, comparisonConfig);
+
+            return result;
+        }
+
+
+        /***************************************************/
+        /****              Private methods              ****/
+        /***************************************************/
+
+        [Description("Returns a Dictionary whose Key is the full property path of the parameter (the FullName of the property holding the parameter), " +
+            "and whose Value is the RevitParameter.")]
+        private static Dictionary<string, RevitParameter> ParametersDictionary<T>(this IBHoMObject obj) where T : class, IRevitParameterFragment
+        {
+            Dictionary<string, RevitParameter> result = new Dictionary<string, RevitParameter>();
+
+            if (!obj?.Fragments?.Any() ?? true)
+                return result;
+
+            string objFullName = obj.GetType().FullName;
+
+            for (int i = 0; i < obj.Fragments.Count(); i++)
+            {
+                T revitParams = obj.Fragments[i] as T;
+                if (revitParams != null)
```

Comment:
No it can't, because we want to get the full-indexed path of the Fragment. So we need to iterate the fragments and retrieve the index of the Parameter in the Fragments list, hence this method. This is driven by consistency with the rest of the diffing, as per comment here: https://github.com/BHoM/Revit_Toolkit/pull/1176#discussion_r819475537